vite/msdfGenerator.js

```diff
@@ -3,6 +3,93 @@ import * as fs from 'fs'
 import { genFont, setGeneratePaths } from '@lightningjs/msdf-generator'
 import { adjustFont } from '@lightningjs/msdf-generator/adjustFont'
 
+// since font generation methods changed with the new renderer version,
+// we need to clean up all the old font files generated by previous plugin implementation
+// this is a temporary solution and should be removed in the future
+let cleanupFinished = false
+let filesToWarn = []
+const dirsToCheck = [
+  {
+    dir: path.resolve('.', 'node_modules', '.tmp-msdf-fonts'),
+    recursive: false,
+    deleteFile: true,
+  },
+  {
+    dir: path.resolve('.', 'public'),
+    recursive: true,
+    deleteFile: false,
+  },
+]
+
+const deleteOldFontFiles = (dir, recursive, deleteFile = false) => {
+  console.log(`Checking ${dir} for old font files...`)
+  const files = fs.readdirSync(dir, { withFileTypes: true })
+
+  files.forEach((file) => {
+    if (file.isDirectory() && recursive) {
+      deleteOldFontFiles(path.join(dir, file.name), recursive, deleteFile)
+    } else {
+      if (file.name.match(/\.json$/i)) {
+        const jsonFile = path.join(dir, file.name)
+        const jsonContent = JSON.parse(fs.readFileSync(jsonFile, 'utf8'))
+
+        if (jsonContent.distanceField && jsonContent.kernings) {
+          const pngFile = path.join(dir, file.name.replace(/\.json$/, '.png'))
+
+          if (!jsonContent.lightningMetrics) {
+            if (deleteFile) {
+              console.log(`Deleting automatically generated old font file: ${jsonFile}`)
+              console.log(`Deleting also the corresponding png file: ${pngFile}`)
+              fs.unlinkSync(jsonFile)
+              fs.unlinkSync(pngFile)
+            } else {
+              filesToWarn.push(jsonFile, pngFile)
+            }
+          }
+        }
+      }
+    }
+  })
+}
+
+const cleanup = () => {
+  if (fs.existsSync(path.resolve('.', '.msdf-cleanup'))) {
```

Comment:
@michielvandergeest I am not sure about having this. If people miss the warning on the first time, they can't see it again unless the temp file is removed. On the other hand, removing this will cause scanning all paths again on each `npm run dev`.